doc/keynames/README.md

```diff
@@ -0,0 +1,477 @@
+According to above rules `system:` would be the Escaped Name of the Root Key for the Namespace "System" and the Escaped Name for the Root Key of the Namespace "Cascading" would be an empty string.
+But this is not the case.
+We use `system:/` and `/` instead.
+The exact details will be explored later, but for now just remember, that an (Escaped) Key Name **always** contains at least one Part Separator (`/`).
+
+But there is a problem.
+We said a Key Name part is "an arbitrary sequence of non-zero bytes".
+This means "elektra/version" is a Key Name Part as well.
+Since the slash `/` would conflict with the Part Separator, we can escape it with a backslash `\` (and `\` can be escaped with another `\`):
+
+```
+/elektra\/version\\/info
+```
+
+This can be deconstructed into:
+
+- A Part Separator: `/`
+- A Key Name Part: `elektra/version\`
+- A Part Separator: `/`
+- A Key Name Part: `info`
+- An invisible null terminator
+
+> _Note:_ When talking about a single Key Name Part `/` and `\` are never escaped.
+
+Because of this escaping mechanism, we call this the _Escaped Name_ of a Key.
+
+Elektra's Key Names are designed, to mimic UNIX paths to an extent.
+To this end we support the commonly used `/.` and `/..`.
+This is one reason, why we need to differentiate between _Canonical_ Key Names and _Non-Canonical_ Key Names.
+
+### 2.1. (Non-)Canonical (Escaped) Key Names
+
+Following the syntax of UNIX paths, in Elektra both `/elektra/./version` and `/elektra/version` refer to the same Key.
+Similarly, `/elektra/../version` and `/version` refer to the same Key.
+
+To give each Key a unique Key Name, we need to introduce a _Canonical (Escaped) Key Name_.
+For UNIX paths, we could say that the canonical path is the shortest possible path that refers to a file.
+In Elektra this doesn't quite work, but will use this definition for now.
+
+> _Note:_ Only Escaped Key Names can be Canonical or Non-Canonical, so we normally omit the "Escaped" specifier.
+
+Let's look at a few examples to get a feeling for Canonical and Non-Canonical Key Names.
+
+| Non-Canonical           | Canonical          |
+| ----------------------- | ------------------ |
+| `/elektra/./version`    | `/elektra/version` |
+| `/elektra/../version`   | `/version`         |
+| `/elektra/.././version` | `/version`         |
+| `/elektra///version`    | `/elektra/version` |
+| `/elektra//../version`  | `/version`         |
+| `/elektra/./../version` | `/elektra/version` |
+| `/elektra/version/`     | `/elektra/version` |
+
+So far this mostly follows, what we know from UNIX paths.
+However, the last two examples are somewhat different.
+
+First, in UNIX `/elektra/./../version` would be the same as `/version`, because `.` is just skipped and `..` removes `elektra`.
+But in Elektra, we evaluate `..` first and only then skip `.`.
+`/elektra/./../version` is the same as `/elektra/version`, because `..` remove `.`.
```

Comment:
@markus2330 @mpranj Do we want to keep that, or should we behave like UNIX paths? AFAIK what is described here was also the behaviour before the PR (I'd have to check to be sure).

Reply:
Yes, sounds reasonable to change to the behavior towards UNIX paths.

Reply:
Then we should also probably change that `/../../..` is a valid path (root stays root).